### PR TITLE
CIF-2222: Add config for storefront context render

### PR DESCRIPTION
**Pull request by laurentiumagureanu**
## Description

Render storefront context data based on config

## Related Issue

CIF-2222

## Motivation and Context

product recs

## How Has This Been Tested?

unit tests

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes and the overall coverage did not decrease.
- [x] All unit tests pass on CircleCi.
- [x] I ran all tests locally and they pass.

### CODE REVIEW

File: bundles/core/src/main/java/com/adobe/cq/commerce/core/components/internal/storefrontcontext/AbstractCommerceStorefrontContext.java [hunk 1]
```diff
@@ -13,25 +13,55 @@
...
+        }
+
+        return storefrontContextEnabled;
+    }
 
     @Override
     public String getJson() {
-        try {
-            return new ObjectMapper().writeValueAsString(this);
+        if (isStorefrontContextEnabled()) {
```
Comment (Buuhuu) [inline bundles/core/src/main/java/com/adobe/cq/commerce/core/components/internal/storefrontcontext/AbstractCommerceStorefrontContext.java:56]: Is there a test that covers the `isStorefrontContextEnabled() = false` case?

**Comment by herzog31** [comment]
LGTM

As @Buuhuu mentioned, please add a test case to test the behavior if the setting is disabled.

File: bundles/core/src/main/java/com/adobe/cq/commerce/core/components/internal/storefrontcontext/AbstractCommerceStorefrontContext.java (hunk 1 above)
Comment (laurentiumagureanu) [reply bundles/core/src/main/java/com/adobe/cq/commerce/core/components/internal/storefrontcontext/AbstractCommerceStorefrontContext.java:56]: added now

**Comment by laurentiumagureanu** [comment]
> LGTM
> 
> As @Buuhuu mentioned, please add a test case to test the behavior if the setting is disabled.

added now

**Review by herzog31 — APPROVED** [review]

**Review by Buuhuu — APPROVED** [review]

**Pull request merged**